x-pack/plugins/xpack_legacy/server/routes/settings.ts

```diff
@@ -42,6 +42,9 @@ export function registerSettingsRoute({
     },
     async (context, req, res) => {
       const { callAsCurrentUser } = context.core.elasticsearch.legacy.client;
+      const collectorFetchClients = {
+        callCluster: callAsCurrentUser,
+      };
```

Comment:
I think this suggestion fixes the question in https://github.com/elastic/kibana/pull/79595#discussion_r500684291

```suggestion
      const collectorFetchClients = {
        callCluster: callAsCurrentUser,
        esClient: context.core.elasticsearch.client.asCurrentUser,
      };
```

Reply:
I went down that avenue yesterday and tried again today but keep hitting the 500 errors [here](https://github.com/elastic/kibana/pull/79595/files/2cad22dc79acb4958d78e0f84d3763fa8049d0ab#diff-9daa2dbdf6ef2c46b15d9e765da1c7bdR53). That means we're not finding the `kibana_settings` collector in the collectors that are registered in the `collector_set` using [`getCollectorByType`](https://github.com/elastic/kibana/blob/master/src/plugins/usage_collection/server/collector/collector_set.ts#L70). The `kibana_settings` collector is a monitoring collector and we haven't added the `esClient` to any of those collectors yet.

Reply:
@joshdover do you have idea why we'd be getting a 500 response from the route handler when adding `esClient: context.core.elasticsearch.client.asCurrentUser`? AFAIK, we can access the new elasticsearch client off of the request context. What I'm not sure about though is how to get around wanting to access both clients on the route.

Reply:
@TinaHeiligers I'm running it locally with the suggested changes and it works fine.

> The `kibana_settings` collector is a monitoring collector

Do you have the `monitoring` plugin disabled? No need for the actual monitoring collection to be enabled, only the plugin must be enabled (xpack's default config).

> and we haven't added the `esClient` to any of those collectors yet.

The `kibana_settings` collector doesn't use any of the ES clients. It only reports what it has in the config of the `monitoring` plugin. We are only sending both clients because the `collector.fetch` method expects them.

If you want, we can do a sync session to debug it together.